lochness/box/__init__.py

```diff
@@ -368,83 +489,108 @@ def sync_module(Lochness: 'lochness.config',
         # loop through the items defined for the BOX data
         for datatype, products in iter(
                 Lochness['box'][module_basename]['file_patterns'].items()):
+            # eg) products
+            # [{'product': 'open',
+            #   'data_dir': 'PronetYA_Interviews/OPEN',
+            #   'out_dir': 'open', ...}, 
+            #  {'product': 'psychs',
+            #  'data_dir': 'PronetYA_Interviews/PSYCHS',
+            #  'out_dir': 'psychs', ...}]
+            for product in products:
+                if Lochness['BIDS']:
+                    datatype_root_obj = get_box_object_based_on_name(
+                            client, product['data_dir'], bx_base_obj.id)
```

Comment:
This file has hardly any change other than `datatype` replaced by `product['data_dir']`. Please double check if the replacement is right.